Fix bug with MIME type warning on good labels

qchaupds opened the pull request:
1. Add test resources : src/test/resources/github429
2. Add two missing types 'DOC' and 'DOCX' for 'Microsoft Word' and 'XLS' and 'XLSX' for 'Microsoft Excel' to mime table src/main/resources/validate_default_mime_types.txt
3. Add new regression test : src/test/resources/features/validate.feature

Refs:

https://github.com/nasa-pds/validate/issues/429 validate warns "document standard id ... is not correct" on good labels

Closes https://github.com/nasa-pds/validate/issues/429 validate warns "document standard id ... is not correct" on good labels

Several missing MIME types for "Microsoft Word" were added to validate_default_mime_types.txt to fix the issue: DOC and DOCX
Several missing MIME types for "Microsoft Excel" were added to validate_default_mime_types.txt to fix the issue: XLS and XLSX

Tested in DEV:

Running validate against a valid label with three document types '.DOC', '.TXT' and '.XLS'.  No warning messages are expected.

```
% validate -R pds4.label -r report_github429_label_valid.json   -s json -t  src/test/resources/github429/EPPS_EDR_SIS.xml
 
```

Commit:
Fix bug with MIME type warning on good labels

1. Add test resources : src/test/resources/github429
2. Add two missing types 'DOC' and 'DOCX' for 'Microsoft Word' and 'XLS' and 'XLSX' for 'Microsoft Excel' to mime table src/main/resources/validate_default_mime_types.txt
3. Add new regression test : src/test/resources/features/validate.feature

Refs:

https://github.com/nasa-pds/validate/issues/429 validate warns "document standard id ... is not correct" on good labels

Commit:
More fixes to fix bug with MIME type warning on good labels

1. Add a bad label : src/test/resources/github429/EPPS_EDR_SIS_bad.xml
2. Add an empty resource : src/test/resources/github429/EPPS_EDR_SIS.txt
3. Add call to getPossibleFileExtensions() to get possible file extensions for a given document id : src/main/java/gov/nasa/pds/tools/util/DocumentsChecker.java
4. Add function getPossibleFileExtensions() to get possible file extensions for a given document id and ignore case when checking for file extension : src/main/java/gov/nasa/pds/tools/util/MimeTable.java
5. Modify error message to include possible file extensions : src/main/java/gov/nasa/pds/tools/validate/rule/pds4/FileReferenceValidationRule.java
6. Add more missing types : src/main/resources/validate_default_mime_types.txt

Refs:

https://github.com/nasa-pds/validate/issues/429 validate warns "document standard id ... is not correct" on good labels

jordanpadams reviewed (changes requested):
@qchaupds i'm not quite seeing the appropriate warnings. when i run against target EPPS_EDR_SIS.xml , I get these warning messages, and I had a couple questions:

* is this message saying EPPS_EDR_SIS.DOC is not really a Word Doc? the file suffix seems valid?
* `Expected file suffix(es): []` - is this supposed to indicate something?
* Can you take a look at the description for #429? I think there are a few sub-tasks I included there that have note been updated?

> Product Level Validation Results
> 
>   PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github429/EPPS_EDR_SIS.xml
>       WARNING  [warning.file.not_msword_mimetype]   file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github429/EPPS_EDR_SIS.DOC with document standard id 'Microsoft Word' is not correct.  Expected file suffix(es): []
>       WARNING  [warning.file.not_msexcel_mimetype]   file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github429/EPPS_EDR_SIS.XLS with document standard id 'Microsoft Excel' is not correct.  Expected file suffix(es): []
>         1 product validation(s) completed

jordanpadams committed:
Update to remove unnecessary error message noise

Commit:
Add more empty resources with different names to avoid connfusion on Mac platform

Commit:
Add more empty resources with different names to avoid connfusion on Mac platform

Commit:
Remove extraneous file_name tags

jordanpadams committed:
Cleanup test data

jordanpadams committed:
Fix resource loading for mime type file

jordanpadams committed:
Cleanup code and handle file name checks

* Removed duplicate map for filenames
* Simplified algorithm

jordanpadams reviewed (changes requested):
@qchaupds I provided some updates to the code and it now builds and executes correctly for the most part except PDF because the map you are using with the text file overwrites.

please checkout the updates I've made and test out on your own.

I would recommend re-evaluating using a text file, and instead use a JSON file with a dictionary that can be read in and maybe flipped around, e.g.

```
{ 
  standard1: [ extension1, extension2, etc ],
  standard2: [ extension1, extension2, etc ]
}
```

```
  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github429/EPPS_EDR_SIS.xml
      WARNING  [warning.file.not_html_mimetype]   file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github429/EPPS_EDR_SIS_1.foo with document standard 'PDF' is not correct.  Expected file suffix(es): []
        1 product validation(s) completed
```

Commit:
Break up github367 test into multiple tests for accounting purposes, and fix bug with finding ProblemType based on a docType

1. Add a good PDF file for testing bad label : src/test/resources/github429/c2h4_Icarus-14111_A.pdf
2. Add initialize() and getProblemType() functions : src/main/java/gov/nasa/pds/tools/util/DocumentUtil.java
3. Add uri to errorMessage : src/main/java/gov/nasa/pds/tools/util/PDFUtil.java
4. Fix bug by modifying logic to find a ProblemType based on a docType, remove commented out codes, only print external error filename if exist : src/main/java/gov/nasa/pds/tools/validate/rule/pds4/FileReferenceValidationRule.java
5. Add more problem types to behavior test code to match ProblemType in FileReferenceValidationRule class : src/test/java/cucumber/StepDefs.java
   (Note that this class controls the Behavior Testing for Cucumber)
6. Fix bug with github366 test, and break up github367 into multiple tests for accounting purposes  : src/test/resources/features/validate.feature
7. Add more document types for more testing bad "7-Bit ASCII Text", "Rich Text" and "MPEG-4" types : src/test/resources/github367/document/pccref.xml
8. Correct file extension of TXT to txt to match actual file on disk (Mac does not distinguish between cases)  : src/test/resources/github429/EPPS_EDR_SIS.xml
9. Add bad label for additional testing of a bad label : src/test/resources/github429/EPPS_EDR_SIS_bad.xml

Refs:

https://github.com/nasa-pds/validate/issues/429 validate warns "document standard id ... is not correct" on good labels

qchaupds committed:
Merge branch 'main' into #429

qchaupds commented on src/main/java/gov/nasa/pds/tools/util/DocumentUtil.java:
```diff
@@ -37,12 +41,109 @@
+      // if we will be using the getProblemType() function below.
+  }
+
+  private boolean isClassInitialized() {
```
@sonatype-lift ignore

jordanpadams committed:
Include rich text as other generic text file

Also includes some code cleanup

jordanpadams committed:
sonatype cleanup

jordanpadams committed:
fix broken test
